packages/js/product-editor/src/blocks/section/edit.tsx

```diff
@@ -15,28 +17,91 @@ import {
  */
 import { sanitizeHTML } from '../../utils/sanitize-html';
 import { SectionBlockAttributes } from './types';
+import { hasAttributesUsedForVariations } from '../../utils';
 
 export function Edit( {
 	attributes,
-}: BlockEditProps< SectionBlockAttributes > ) {
+	context,
+}: BlockEditProps< SectionBlockAttributes > & {
+	context?: {
+		selectedTab?: string | null;
+	};
+} ) {
 	const { description, title, blockGap } = attributes;
+	const tab = context?.selectedTab || '';
 	const blockProps = useBlockProps();
+	const [ productAttributes ] = useEntityProp< Product[ 'attributes' ] >(
+		'postType',
+		'product',
+		'attributes'
+	);
+	const isDisabledSection =
+		hasAttributesUsedForVariations( productAttributes ) &&
+		[ 'inventory', 'pricing', 'shipping' ].includes( tab );
 	const innerBlockProps = useInnerBlocksProps(
 		{
 			className: classNames(
 				'wp-block-woocommerce-product-section__content',
-				`wp-block-woocommerce-product-section__content--block-gap-${ blockGap }`
+				`wp-block-woocommerce-product-section__content--block-gap-${ blockGap }`,
+				{
+					'is-disabled-section': isDisabledSection,
+				}
 			),
 		},
 		{ templateLock: 'all' }
 	);
 	const SectionTagName = title ? 'fieldset' : 'div';
 	const HeadingTagName = SectionTagName === 'fieldset' ? 'legend' : 'div';
 
+	useEffect( () => {
+		const elementsToReEnable: (
+			| HTMLInputElement
+			| HTMLButtonElement
+			| HTMLSelectElement
+			| HTMLTextAreaElement
+		 )[] = [];
+
+		if ( isDisabledSection ) {
+			const disabledSections = document.querySelectorAll(
+				'.is-disabled-section'
+			);
+
+			disabledSections.forEach( ( section ) => {
+				const controls = section.querySelectorAll(
+					'input, button, select, textarea'
+				);
+
+				controls.forEach( ( control ) => {
+					if (
+						control instanceof HTMLInputElement ||
+						control instanceof HTMLButtonElement ||
+						control instanceof HTMLSelectElement ||
+						control instanceof HTMLTextAreaElement
+					) {
+						if ( ! control.disabled ) {
+							elementsToReEnable.push( control );
+							control.disabled = true;
+						}
+					}
+				} );
+			} );
+		}
+		return () => {
+			elementsToReEnable.forEach( ( control ) => {
+				control.disabled = false;
+			} );
```

Comment:
I wonder if there would be any problem if some of the elements were no longer available (again, this is why I am against having block implementation knowledge instead of the section). Because, we have no guarantee that a block wouldn't change it's render since we initially disabled some of its elements.

Reply:
I'm not sure if I understood the point here. The cleanup function will remove the disabled prop from the elements when we unmount the section component.